wikimedia/wikipedia-ios
# Add Notifications Bell button to navigation bar

**Phabricator:** https://phabricator.wikimedia.org/T288652

### Notes
- Adds the Notifications bell assets
- Updates the navigation bar to add new display type to support centered title views, but still retaining existing large title behavior. The existing `largeTitle` type always adds the title view left aligned to the navigation bar, which isn't the desired presentation we're going for when the notifications bell button is available.
- Add notifications bar button to Explore feed and to Settings view when Explore feed is off

### Test Steps
At this point, we’re not concerned about seeing odd states of appearance of the notifications bell in the Explore feed/Settings view. We just want to confirm it renders in the expected contexts (Explore feed and Settings with Explore feed off) and that the new `NavigationBar` display type has no regressions. The proper logic to update the bell’s indicated or un-indicated image, as well as whether it’s displayed or not based on user authentication status will be covered by: [T288669](https://phabricator.wikimedia.org/T288669). Some code (like `notificationsCenterBellImageWithUnreadNotifications:`)  is included for demonstration and will move or be removed as new tasks proceed.

1. Confirm building and running displays the Notifications bell as the left most Explore feed bar button
2. Comment out ` _exploreViewController.navigationItem.leftBarButtonItem = notificationsBarButton;` [here](https://github.com/wikimedia/wikipedia-ios/blob/39c31545c591e30f0aba7aeaa92557e8887eff0a/Wikipedia/Code/WMFAppViewController.m#L1379) – confirm the Explore feed navigation bar renders as it does before this PR (left aligned title view, right aligned settings bar button)
3. Update [this](https://github.com/wikimedia/wikipedia-ios/blob/39c31545c591e30f0aba7aeaa92557e8887eff0a/Wikipedia/Code/WMFAppViewController.m#L1374) to render the alternate bell image: `[self notificationsCenterBellImageWithUnreadNotifications:NO]` – confirm the bell image without the indicator renders

### Screenshots

![NotificationsBell](https://user-images.githubusercontent.com/5652327/129984092-659b7792-1528-4b14-b23b-30efb29374b7.png)
